tests/integration/serialization/test_serializers.py

```diff
@@ -0,0 +1,319 @@
+                          (ShortSerializer(), random.randint(-32768, 32768)),
+                          (DoubleSerializer(), None),
+                          (LongSerializer(), None),
+                          (IntegerSerializer(), None),
+                          (ShortSerializer(), None)])
+def test_numeric_serialization(kafka_cluster, serializer, data):
+    """
+    Tests basic serialization/deserialization of numeric types.
+
+    :param KafkaClusterFixture kafka_cluster: cluster fixture
+    :param Serializer serializer: serializer to test
+    :param object data: input data
+
+    :raises: AssertionError on test failure
+
+    :returns: None
+    :rtype: None
+    """
+    topic = kafka_cluster.create_topic("serialization-numeric")
+
+    producer = kafka_cluster.producer(value_serializer=serializer)
+    producer.produce(topic, data)
+    producer.flush()
+
+    consumer = kafka_cluster.consumer(value_serializer=serializer)
+    consumer.subscribe([topic])
+
+    msg = consumer.poll()
+
+    assert msg.value() == data
+
+    consumer.close()
+
+
+@pytest.mark.parametrize("data",
+                         [random.uniform(-1.0, 1.0),
+                          None])
+def test_float_serialization(kafka_cluster, data):
+    """
+    Tests basic float serialization/deserialization functionality.
+
+    This test must be separated from the standard numeric types due
+    to the nature of floats in Python. Some precision loss occurs
+    when converting python's double-precision float to the expected
+    single-precision float value. As such we only test that the
+    single-precision approximated values is reasonably close to the original.
+
+    :param KafkaClusterFixture kafka_cluster: cluster fixture
+    :param float data: test data
+
+    :raises: AssertionError on test failure
+
+    :returns: None
+    :rtype: None
+    """
+    topic = kafka_cluster.create_topic("serialization-float")
+
+    producer = kafka_cluster.producer(key_serializer=FloatSerializer())
+
+    data = random.uniform(-1.0, 1.0)
+    producer.produce(topic, key=data)
+    producer.flush()
+
+    consumer = kafka_cluster.consumer(key_serializer=FloatSerializer())
+    consumer.subscribe([topic])
+
+    msg = consumer.poll()
+
+    assert data == pytest.approx(msg.key())
+
+    consumer.close()
+
+
+@pytest.mark.parametrize("data, codec",
+                         [(u'Jämtland', 'utf_8'),
+                          (u'Härjedalen', 'utf_16'),
+                          (None, 'utf_32')])
+def test_string_serialization(kafka_cluster, data, codec):
+    """
+    Tests basic unicode serialization/deserialization functionality
+
+    :param KafkaClusterFixture kafka_cluster: cluster fixture
+    :param unicode data: input data
+    :param str codec: encoding type
+
+    :raises: AssertionError on test failure
+
+    :returns: None
+    :rtype: None
+    """
+    topic = kafka_cluster.create_topic("serialization-string")
```

Comment:
Use a testtopic prefix, and preferably a random part to make test-re-runs dependible.

Reply:
Unless I'm misunderstanding your suggestion I think I already do this 

kafka_cluster.create_topic 
```
    def create_topic(self, prefix, conf=None):
        """
        Creates a new topic with this cluster.

        :param str prefix: topic name
        :param dict conf: additions/overrides to topic configuration.
        :returns: The topic's name
        :rtype: str
        """
        if self._admin is None:
            self._admin = AdminClient(self.client_conf())

        name = prefix + str(uuid1())
        future_topic = self._admin.create_topics([NewTopic(name,
                                                           **self._topic_conf(conf))])

        future_topic.get(name).result()
        return name
```